lib/yardstick/decorator.rb

```diff
@@ -0,0 +1,55 @@
+module Yardstick
+  # A string decorator for applying unix console codes
+  class Decorator
+    include Concord.new(:color, :mode)
+
+    FORMAT = "\e[%{mode};%{color}m%{string}\e[0m".freeze
+
+    COLOR_CODES = {
```

Comment:
I'd just use `IceNine.deep_freeze` for Hashes, regardless of what the Hashes contain. IceNine skips over anything it is unable to freeze.